internals/cli/cmd_add_test.go

```diff
@@ -100,12 +100,15 @@ services:
 			} else if path == triggerLayerPath {
 				c.Assert(err, check.ErrorMatches, "triggered")
 			} else if path == unreadableLayerPath {
-				c.Assert(os.IsPermission(err), check.Equals, true)
+				if os.Getuid() != 0 {
+					c.Assert(os.IsPermission(err), check.Equals, true)
+				}
 			}
 		}
 
 		args = append(args, "extra", "arguments", "invalid")
 		_, err = cli.ParserForTest().ParseArgs(args)
 		c.Assert(err, check.Equals, cli.ErrExtraArgs)
+		s.ResetStdStreams()
```

Comment:
This test case uses a for loop to test `combine=false` and `combine=true`.

When running as root, adding the `unreadableLayer` would succeed and the stdout interferes with the next loop. So, here I added a reset on the stdout so that the two cases don't interfere with each other.